manifests/init.pp

```diff
@@ -363,46 +430,63 @@
 #                                     type:boolean
 #
 # $server_foreman_url::               Foreman URL
+#                                     type:string
 #
 # $server_foreman_ssl_ca::            SSL CA of the Foreman server
+#                                     type:string
 #
 # $server_foreman_ssl_cert::          Client certificate for authenticating against Foreman server
+#                                     type:string
 #
 # $server_foreman_ssl_key::           Key for authenticating against Foreman server
+#                                     type:string
 #
 # $server_puppet_basedir::            Where is the puppet code base located
+#                                     type:string
 #
 # $server_enc_api::                   What version of enc script to deploy. Valid
 #                                     values are 'v2' for latest, and 'v1'
 #                                     for Foreman =< 1.2
+#                                     type:string
 #
 # $server_report_api::                What version of report processor to deploy.
 #                                     Valid values are 'v2' for latest, and 'v1'
 #                                     for Foreman =< 1.2
+#                                     type:string
 #
 # $server_request_timeout::           Timeout in node.rb script for fetching
 #                                     catalog from Foreman (in seconds).
 #                                     type:integer
 #
 # $server_environment_timeout::       Timeout for cached compiled catalogs (10s, 5m, ...)
+#                                     type:string
 #
 # $server_ca_proxy::                  The actual server that handles puppet CA.
 #                                     Setting this to anything non-empty causes
 #                                     the apache vhost to set up a proxy for all
 #                                     certificates pointing to the value.
+#                                     type:string
 #
 # $server_jvm_java_bin::              Set the default java to use.
+#                                     type:string
 #
 # $server_jvm_config::                Specify the puppetserver jvm configuration file.
+#                                     type:string
 #
 # $server_jvm_min_heap_size::         Specify the minimum jvm heap space.
+#                                     type:string
 #
 # $server_jvm_max_heap_size::         Specify the maximum jvm heap space.
+#                                     type:string
 #
 # $server_jvm_extra_args::            Additional java options to pass through.
 #                                     This can be used for Java versions prior to
 #                                     Java 8 to specify the max perm space to use:
 #                                     For example: '-XX:MaxPermSpace=128m'.
+#                                     type:array
```

Comment:
string also here, although this could be changed, I'd leave it as string for now, this can get refactored later, if needed.